src/kv/kvtypes.h

```diff
@@ -5,13 +5,16 @@
 #include "crypto/hash.h"
 
 #include <array>
+#include <chrono>
 #include <functional>
 #include <limits>
 #include <memory>
+#include <unordered_set>
 #include <vector>
 
 namespace kv
 {
+  using Index = int64_t;
```

Comment:
What's an Index? (How does it differ from a Version?)

Reply:
(Looks like its only used in Replicator, could it be `Replicator::ConfigurationIndex`?)

Reply:
Index is the Raft terminology (I almost wrote "term"!), as opposed to Version, which is the KV equivalent.

Reply:
Yes that was copied over from raft.h. I'll change it to ConfigIndex

Reply:
Why config index? It should be the generic consensus term we adopt for both Raft and PBFT.

Reply:
add_configuration is a bit awkward right now since it is used both in pbft and in raft and they need different info to configure nodes. Index is only used by raft at the moment. pbft doesn't really have index in it's terminology at the moment.

Reply:
Yep, let's align terminology as agreed.